File: cfgs/java_cfgs_foo/REMOVE_ARRAY_END_ELEMENT_MAXIMIZE_SUM_PRODUCT.java
Nodes
1;special;;;
2;conditional;foo_1 == foo_2;14;
3;statement;return foo_1[foo_2] * foo_3;15;
4;exit;;;
5;conditional;foo_1[foo_2][foo_3] != 0;17;
6;statement;return foo_1[foo_2][foo_3];18;
7;statement;foo_1[foo_2][foo_3] = fun_1((foo_4[foo_5] * foo_6) + fun_2(foo_7, foo_8, foo_9 + 1, foo_10, foo_11 + 1), (foo_12[foo_13] * foo_14) + fun_3(foo_15, foo_16, foo_17, foo_18 - 1, foo_19 + 1));20;
8;statement;return foo_1[foo_2][foo_3];21;
Edges
0;2;
1;3;5;
2;4;
3;
4;6;7;
5;4;
6;8;
7;4;

File: cfgs/pythoncfgs/REMOVE_ARRAY_END_ELEMENT_MAXIMIZE_SUM_PRODUCT.py
Nodes
1;conditional;foo_1 == foo_2;8;
2;conditional;foo_1[foo_2][foo_3] != 0;10;
3;statement;foo_1[foo_2][foo_3] = fun_3(foo_4[foo_5] * foo_6 + fun_2(foo_7, foo_8, foo_9 + 1, foo_10, foo_11 + 1), foo_12[foo_13] * foo_14 + fun_3(foo_15, foo_16, foo_17, foo_18 - 1, foo_19 + 1));12;
4;statement;return foo_1[foo_2][foo_3];13;
5;statement;return foo_1[foo_2][foo_3];11;
6;statement;return foo_1[foo_2] * foo_3;9;
7;exit;;
Edges
0;2;6;
1;3;5;
2;4;
3;7;
4;7;
5;7;
6;